Arrowgene.Ddon.GameServer/Characters/EquipManager.cs

```diff
@@ -104,9 +121,65 @@ public void HandleChangeEquipList(DdonGameServer server, GameClient client, Char
                     // Update storage, swapping if needed
                     Storage sourceStorage = client.Character.Storage.GetStorage(storageTypes[0]);
                     updateCharacterItemNtc.UpdateItemList.AddRange(server.ItemManager.MoveItem(server, client.Character, sourceStorage, itemUId, 1, characterToEquipTo.Equipment.Storage, equipItemStorageSlot));
+
+                    //Check for costume requirements
+                    ClientItemInfo itemInfo = server.ItemManager.LookupInfoByUID(server, itemUId);
+                    if (itemInfo.SubCategory == ItemSubCategory.EquipEnsemble)
+                    {
+                        foreach (EquipSlot slot in EnsembleSlots)
+                        {
+                            var foo = characterToEquipTo.EquipmentTemplate.GetEquipItem(characterToEquipTo.Job, equipType, (byte)slot);
+                            if (foo is null) continue;
+                            forceRemovals.Add((equipType, slot));
+                        }
+                    }
+                    else if (EnsembleSlots.Contains((EquipSlot)(itemInfo.SubCategory - 300))) //Casting ItemSubCategory -> EquipSlot.
```

Comment:
This substraction is a bit obtuse out of context, it may be better to have a named constant for the 300, or a comment explaining the reasoning. Keep in mind this code has been rewritten a thousand times and i dont think this'll be the last time, so it's going to be useful for whoever next rewrites it to clearly understand what the purpose of this is